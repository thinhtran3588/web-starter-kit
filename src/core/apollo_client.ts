import ApolloClient, { gql } from 'apollo-boost';
import fetch from 'isomorphic-unfetch';
import { config } from '@app/config';

export const apolloClient = new ApolloClient({
  uri: config.apiEndpoint,
  fetch,
});
export { gql };
