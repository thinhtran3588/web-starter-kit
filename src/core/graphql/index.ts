export * from './queries/getCurrentUserQuery';
export * from './queries/getCurrentNotificationQuery';
export * from './apolloClient';
