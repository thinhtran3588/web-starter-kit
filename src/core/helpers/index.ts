export * from './format_with_lookup';
export * from './format_date';
export * from './format_date_time';
export * from './show_notification';
export * from './catch_error';
export * from './handle_error';
export * from './app_error';
export * from './sanitize_form_data';
export * from './get_error_message';
export * from './sleep';
