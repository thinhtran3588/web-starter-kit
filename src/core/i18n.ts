import NextI18Next, {
  WithTranslation as NextI18NextWithTranslation,
  TFunction as NextI18NexTFunction,
} from 'next-i18next';
import { config } from '@app/config';

const otherLanguages = config.i18n.languages
  .filter((language) => language.code !== config.i18n.defaultLang)
  .map((language) => language.code);
const localeSubpaths: { [code: string]: string } = {};
otherLanguages.forEach((language) => {
  localeSubpaths[language] = language;
});
export const nextI18next = new NextI18Next({
  defaultLanguage: config.i18n.defaultLang,
  otherLanguages,
  localeSubpaths,
});

/* Optionally, export class methods as named exports */
export const { appWithTranslation, withTranslation, i18n } = nextI18next;
export type WithTranslation = NextI18NextWithTranslation;
export type TFunction = NextI18NexTFunction;
