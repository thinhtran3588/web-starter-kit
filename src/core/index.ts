export * from './theme';
export * from './interfaces/nav_item';
export * from './interfaces/offset_pagination';
export * from './interfaces/offset_pagination_result';
export * from './interfaces/field_info';
export * from './interfaces/filter';
export * from './interfaces/filte_with_offset_pagination';
export * from './interfaces/picker_data_item';
export * from './interfaces/field_type';
export * from './interfaces/field_value_type';
export * from './interfaces/table_column';
export * from './interfaces/next_request';
export * from './interfaces/search_record';
export * from './i18n';
export * from './apollo_client';
