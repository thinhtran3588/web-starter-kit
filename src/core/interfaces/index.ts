export * from './nav_item';
export * from './offset_pagination';
export * from './offset_pagination_result';
export * from './field_info';
export * from './filter';
export * from './filte_with_offset_pagination';
export * from './picker_data_item';
export * from './field_type';
export * from './field_value_type';
export * from './table_column';
export * from './next_request';
export * from './search_record';
export * from './user';
export * from './login_type';
export * from './form_data';
export * from './notification_content';
export * from './country';
export * from './error_with_code';
