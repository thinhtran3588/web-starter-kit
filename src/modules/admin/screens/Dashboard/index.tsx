import React from 'react';
import { AdminLayout } from '@app/components';
import { useStyles } from './styles';

export const Dashboard = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AdminLayout title='Dashboard' description='Dashboard'>
      <h1 className={classes.header}>Dashboard</h1>
    </AdminLayout>
  );
};
