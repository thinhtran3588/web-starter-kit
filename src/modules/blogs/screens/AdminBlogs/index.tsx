import React from 'react';
import { AdminLayout } from '@app/components';
import { useStyles } from './styles';

export const AdminBlogs = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AdminLayout title='AdminBlogs' description='AdminBlogs'>
      <h1 className={classes.header}>AdminBlogs</h1>
    </AdminLayout>
  );
};
