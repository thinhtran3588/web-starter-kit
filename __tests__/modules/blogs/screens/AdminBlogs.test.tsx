import React from 'react';
import { render } from '@testing-library/react';
import { AdminBlogs } from '@app/modules/blogs/screens/AdminBlogs';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core';

describe('@app/modules/AdminBlogs/screens/AdminBlogs', () => {
  const theme = createMuiTheme({
    props: {
      MuiWithWidth: {
        initialWidth: 'xs',
      },
    },
  });
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(
      <MuiThemeProvider theme={theme}>
        <AdminBlogs />
      </MuiThemeProvider>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
