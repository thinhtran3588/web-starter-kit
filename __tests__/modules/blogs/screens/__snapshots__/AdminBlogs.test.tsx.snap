// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@app/modules/AdminBlogs/screens/AdminBlogs renders successfully 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-3 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <img
          alt="Logo"
          class="makeStyles-logo-5"
          src="/static/images/logo.svg"
        />
        <div
          class="makeStyles-flexGrow-4"
        />
        <button
          aria-label="notification"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <span
              class="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
              />
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="logout"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          data-testid="menu-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="MuiDrawer-root MuiDrawer-docked"
      data-testid="sidebar"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-drawer-86 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
      >
        <ul
          class="MuiTreeView-root makeStyles-root-99"
          role="tree"
        >
          <li
            aria-expanded="true"
            class="MuiTreeItem-root makeStyles-rootItem-100 Mui-expanded makeStyles-expanded-103"
            role="treeitem"
            tabindex="-1"
          >
            <div
              class="MuiTreeItem-content makeStyles-content-101"
            >
              <div
                class="MuiTreeItem-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <div
                class="MuiTypography-root MuiTreeItem-label makeStyles-label-104 MuiTypography-body1"
              >
                <div
                  class="makeStyles-labelRoot-105"
                  style="color: rgb(26, 115, 232);"
                >
                  <span
                    class="makeStyles-labelIconContainer-106"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-labelIcon-107"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-labelText-108 MuiTypography-body2"
                  >
                    Auth
                  </p>
                </div>
              </div>
            </div>
            <ul
              class="MuiCollapse-container MuiTreeItem-group makeStyles-group-102 MuiCollapse-entered"
              role="group"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <li
                    class="MuiTreeItem-root makeStyles-rootItem-100"
                    role="treeitem"
                    tabindex="-1"
                  >
                    <div
                      class="MuiTreeItem-content makeStyles-content-101"
                    >
                      <div
                        class="MuiTreeItem-iconContainer"
                      >
                        <div
                          style="width: 24px;"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTreeItem-label makeStyles-label-104 MuiTypography-body1"
                      >
                        <a
                          class="makeStyles-link-109 makeStyles-labelRoot-105"
                          href="/admin/users"
                          title="Users"
                        >
                          <div
                            class="makeStyles-labelRoot-105"
                            style="color: rgb(26, 115, 232);"
                          >
                            <span
                              class="makeStyles-labelIconContainer-106"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root makeStyles-labelIcon-107"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                                />
                              </svg>
                            </span>
                            <p
                              class="MuiTypography-root makeStyles-labelText-108 MuiTypography-body2"
                            >
                              Users
                            </p>
                          </div>
                        </a>
                      </div>
                    </div>
                  </li>
                  <li
                    class="MuiTreeItem-root makeStyles-rootItem-100"
                    role="treeitem"
                    tabindex="-1"
                  >
                    <div
                      class="MuiTreeItem-content makeStyles-content-101"
                    >
                      <div
                        class="MuiTreeItem-iconContainer"
                      >
                        <div
                          style="width: 24px;"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTreeItem-label makeStyles-label-104 MuiTypography-body1"
                      >
                        <a
                          class="makeStyles-link-109 makeStyles-labelRoot-105"
                          href="/admin/roles"
                          title="Roles"
                        >
                          <div
                            class="makeStyles-labelRoot-105"
                            style="color: rgb(26, 115, 232);"
                          >
                            <span
                              class="makeStyles-labelIconContainer-106"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root makeStyles-labelIcon-107"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                                />
                              </svg>
                            </span>
                            <p
                              class="MuiTypography-root makeStyles-labelText-108 MuiTypography-body2"
                            >
                              Roles
                            </p>
                          </div>
                        </a>
                      </div>
                    </div>
                  </li>
                </div>
              </div>
            </ul>
          </li>
          <li
            aria-expanded="false"
            class="MuiTreeItem-root makeStyles-rootItem-100"
            role="treeitem"
            tabindex="-1"
          >
            <div
              class="MuiTreeItem-content makeStyles-content-101"
            >
              <div
                class="MuiTreeItem-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 17l5-5-5-5v10z"
                  />
                  <path
                    d="M0 24V0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </div>
              <div
                class="MuiTypography-root MuiTreeItem-label makeStyles-label-104 MuiTypography-body1"
              >
                <div
                  class="makeStyles-labelRoot-105"
                  style="color: rgb(26, 115, 232);"
                >
                  <span
                    class="makeStyles-labelIconContainer-106"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-labelIcon-107"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                      />
                    </svg>
                  </span>
                  <p
                    class="MuiTypography-root makeStyles-labelText-108 MuiTypography-body2"
                  >
                    Public Website
                  </p>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <main
      class="makeStyles-content-2"
    >
      <h1
        class="makeStyles-header-1"
      >
        AdminBlogs
      </h1>
      <footer
        class="makeStyles-footer-152"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          ©
           
          <a
            href="/"
          >
            Thinh Tran
          </a>
           2019
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          Created with love!
        </span>
      </footer>
    </main>
  </div>
</body>
`;
